cmd/block-generator/runner/run.go

```diff
@@ -82,33 +83,140 @@ func (r *Args) run() error {
 	return nil
 }
 
-// Helper to record the import rate.
-func recordDataToFile(entry Entry, key string, out *os.File) error {
-	sum := 0.0
-	count := 0.0
+type metricType int
 
-	for _, metric := range entry.Data {
-		var err error
-		if strings.HasPrefix(metric, "indexer_daemon_import_time_sec_sum") {
-			val := strings.Split(metric, " ")[1]
-			sum, err = strconv.ParseFloat(val, 64)
-		} else if strings.HasPrefix(metric, "indexer_daemon_import_time_sec_count") {
-			val := strings.Split(metric, " ")[1]
-			count, err = strconv.ParseFloat(val, 64)
-		}
-		if err != nil {
-			return fmt.Errorf("unable to parse metric '%s': %w", metric, err)
+const (
+	rate metricType = iota
+	intTotal
+	floatTotal
+)
+
+// Helper to record metrics. Supports rates (sum/count) and counters.
+func recordDataToFile(start time.Time, entry Entry, prefix string, out *os.File) error {
+	num := 0
+	var writeErrors strings.Builder
+	var writeErr error
+	record := func(prefix2, name string, t metricType) {
+		key := fmt.Sprintf("%s%s_%s", prefix, prefix2, name)
+		if err := recordMetricToFile(entry, key, name, t, out); err != nil {
+			writeErr = err
+			if writeErrors.Len() > 0 {
+				writeErrors.WriteString(", ")
+			}
+			writeErrors.WriteString(name)
```

Comment:
honestly, the main reason was to avoid having to check return values after every call.

I do kinda like accumulating errors like this, if two of the metrics were missing it would report both of them instead of just the first.

Reply:
I dunno, I tried changing. Do you think this better?
```
  record := func(prefix2, name string, t metricType) {
    key := fmt.Sprintf("%s%s_%s", prefix, prefix2, name)
    return recordMetricToFile(entry, key, name, t, out)
  }

  if err := record("_average", metrics.BlockImportTimeName, rate); err != nil {    
    return err     
  }    
  if err := record("_cumulative", metrics.BlockImportTimeName, floatTotal); err != nil {    
    return err                                                                                                                                                                                                                                                 
  }                                                                                                                                                                                                                                                            
  if err := record("_average", metrics.ImportedTxnsPerBlockName, rate); err != nil {                                                                                                                                                                           
    return err                                                                                                                                                                                                                                                 
  }                                                                                                                                                                                                                                                            
  if err := record("_cumulative", metrics.ImportedTxnsPerBlockName, intTotal); err != nil {                                                                                                                                                                    
    return err                                                                                                                                                                                                                                                 
  }                                                                                                                                                                                                                                                            
  if err := record("_average", metrics.BlockUploadTimeName, rate); err != nil {                                                                                                                                                                                
    return err                                                                                                                                                                                                                                                 
  }                                                                                                                                                                                                                                                            
  if err := record("_cumulative", metrics.BlockUploadTimeName, floatTotal); err != nil {                                                                                                                                                                       
    return err                                                                                                                                                                                                                                                 
  }                                                                                                                                                                                                                                                            
  if err := record("", metrics.ImportedRoundGaugeName, intTotal); err != nil {                                                                                                                                                                                 
    return err                                                                                                                                                                                                                                                 
  }                                                                                                                                                                                                                                                            
```

Reply:
haha maybe not. This looks more like go :-P 
It's your call. I was curious if there was another reason for it. You can leave it as is if you like.